drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -325,39 +344,85 @@ def process(transaction):
-		fileName = getFileFromLine(line)
+		# # Retrieve the image file name, please no whitespace characters in filename!
+		# fileName = getFileFromLine(line)
+		# # Due to the datahandler we need to add another subfolder of the same name to the path
+		# imageFolder = os.path.join(incomingPath, folderName)
+		# imageFile = os.path.join(imageFolder, fileName)
+		# log_print("New incoming image file for OMERO registration:\t" + imageFile)
+
+		# Retrieve the image folder path, please no whitespace characters in filename!
+		# The string "./" is set to point to the relative root folder
+		relativeFolderPath = getImageFolderPathFromLine(line)
+		if relativeFolderPath == "./":
+			relativeFolderPath = ""
+
+		log_print("RelativeFolderPath: " + relativeFolderPath)
 		# Due to the datahandler we need to add another subfolder of the same name to the path
-		imageFolder = os.path.join(incomingPath, folderName)
-		imageFile = os.path.join(imageFolder, fileName)
-		print "New incoming image file for OMERO registration:\t" + imageFile
+		basePath = os.path.join(incomingPath, folderName)
+		log_print("BasePath: " + basePath)
+		imageFolderPath = os.path.join(basePath, relativeFolderPath)
+		log_print("Incoming folder path for OMERO import: " + imageFolderPath)
+
+		# Look for overriding SAMPLE_ID in line/property map
+		if "SAMPLE_ID" in properties.keys():
+			if len(properties["SAMPLE_ID"]) == 10:
+				line_project_code = properties["SAMPLE_ID"][:5]
+				line_sample_code = properties["SAMPLE_ID"]
+				omero_dataset_id = registrationProcess.requestOmeroDatasetId(project_code=line_project_code, sample_code=line_sample_code)
+			else:
+				omero_dataset_id = default_omero_dataset_id
+		else:
+			omero_dataset_id = default_omero_dataset_id
+		log_print("Iteration omero dataset id: " + str(omero_dataset_id))
+
 
 		# 4. After we have received the omero dataset id, we know where to attach the image to
 		# in OMERO. We pass the omero dataset id and trigger the image registration process in OMERO.
-		omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
-		print "Created OMERO image identifiers:\t" + str(omero_image_ids)
+		# omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
+		omero_image_ids = registrationProcess.registerImageFolder(imageFolderPath, omero_dataset_id)
+		
+		log_print("Created OMERO image identifiers: " + str(omero_image_ids))
 
 		omero_failed = len(omero_image_ids) < 1
 		if omero_failed:
 			raise ValueError("Omero did not return expected image ids.")
 
-		# 5. Additional metadata is provided in an own metadata TSV file. 
-		# We extract the metadata from this file.
-		properties = getPropertyMap(line, property_names)
-
 		# 5.1 Validate metadata for image file
-		ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
+		# ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
```

Comment:
Remove outdatedl ine
```suggestion
```

Reply:
This line I would like to keep, since here I am just temporally disabled metadata validation, pending REMBI alignment and and some ETL logic modifications, but this will not make this cut for the release. This REMBI alignment of ontology vocabularies needs to be done properly, so I will just patch this later once we update the metadata model properly.

Reply:
That's fine however I'd then recommend a ToDo here instead of having the line duplicated and commented out 🤔